src/nix/repl.cc

```diff
@@ -471,7 +471,14 @@ bool NixRepl::processLine(string line)
         auto args = editorFor(pos);
         auto editor = args.front();
         args.pop_front();
-        runProgram(editor, true, args);
+
+        try {
+            // runProgram redirects stdout to a StringSink,
+            // using runProgram2 to allow editors to display their UI
+            runProgram2(RunOptions { .program = editor, .searchPath = true, .args = args });
+        } catch (ExecError & e) {
+            // Ignore errors from editor
```

Comment:
You probably don’t want to catch the error here. It’s caught somewhere above in the stack so that it doesn’t crash the repl but it just printed (that can probably help debugging issues caused by a bad `$EDITOR`, and that’s what `runProgram` was emulating too):

```console
$ EDITOR=false nix repl '<nixpkgs>'
Welcome to Nix 2.5pre20211105_3d6ee22. Type :? for help.

Loading '<nixpkgs>'...
Added 15309 variables.

nix-repl> :e hello
error: program 'false' failed with exit code 1

nix-repl>
```

Reply:
Thanks, good catch!